datafusion/core/src/datasource/listing/table.rs

```diff
@@ -123,9 +127,23 @@ impl ListingTableConfig {
             table_paths: self.table_paths,
             file_schema: self.file_schema,
             options: Some(listing_options),
+            schema_adapter_factory: self.schema_adapter_factory,
         }
     }
 
+    /// Add a schema adapter factory to the [`ListingTableConfig`]
+    ///
+    /// Schema adapters handle schema evolution over time, allowing the table to adapt
+    /// to changes in file schemas. This is particularly useful for handling nested fields
+    /// in formats like Parquet where the schema may evolve.
+    pub fn with_schema_adapter_factory(
```

Comment:
at a high level, it makes a lot of sense to provide the schema adapter factory to the listing table